File: 509: FibonacciNumber/dynamicProgramming.py
class Solution(object):
    def fib(self, n):
        """
        :type n: int
        :rtype: int
        """
        a = 1
        b = 1
        t = 0
        x = 2
        if n == 0 or n == 1:
            return n
        while x<=n:
            t = b
            b = t+a
            a = t
            x+=1
        return t

File: 509: FibonacciNumber/dynamicProgramming.c
int fib(int n){
    int a = 1;
    int b = 1;
    if(n < 2){
        return n;
    }
    n--;
    while(--n>0){
        b +=a;
        a = b-a;
    }
    return b;
}
